src/server/list_family.cc

```
@@ -1173,8 +1173,138 @@ void BPopGeneric(ListDir dir, CmdArgList args, Transaction* tx, SinkReplyBuilder
   return rb->SendNullArray();
 }
 
+// Validates the LMPop command arguments and extracts the values to the output params.
+// If the arguments are invalid sends the appropiate error to builder and returns false.
+bool ValidateLMPopCommand(CmdArgList args, uint32* num_keys, bool* is_left, int* pop_count,
+                          SinkReplyBuilder* builder) {
+  CmdArgParser parser{args};
+
+  if (!SimpleAtoi(parser.Next(), num_keys)) {
+    builder->SendError(kUintErr);
+    return false;
+  }
+
+  if (*num_keys <= 0 || !parser.HasAtLeast(*num_keys + 1)) {
+    // We should have at least num_keys keys + a LEFT/RIGHT arg.
+    builder->SendError(kSyntaxErr);
+    return false;
+  }
+  // Skip over the keys themselves.
+  parser.Skip(*num_keys);
+
+  // We know we have at least one more arg (we checked above).
+  if (parser.Check("LEFT")) {
+    *is_left = true;
+  } else if (parser.Check("RIGHT")) {
+    *is_left = false;
+  } else {
+    builder->SendError(kSyntaxErr);
+    return false;
+  }
+
+  *pop_count = 1;  // Default value if COUNT is not specified
+  // Check if we have additional COUNT argument.
+  if (parser.HasNext()) {
+    if (!parser.Check("COUNT", pop_count)) {
+      builder->SendError(kSyntaxErr);
+      return false;
+    }
+  }
+
+  if (!parser.Finalize()) {
+    builder->SendError(parser.Error()->MakeReply());
+    return false;
+  }
+
+  return true;
+}
+
+// Returns all non-empty keys found in the list of shard arguments
+std::vector<std::string_view> GetNonEmptyKeysFound(EngineShard* shard, Transaction* t) {
+  ShardArgs keys = t->GetShardArgs(shard->shard_id());
+  if (keys.Empty()) {
+    return {};
+  }
+
+  auto& db_slice = t->GetDbSlice(shard->shard_id());
+  std::vector<std::string_view> result;
+
+  for (string_view key : keys) {
+    auto it = db_slice.FindReadOnly(t->GetDbContext(), key, OBJ_LIST);
```

We should include the `key` if it's not `list` and if it's first in the command argument list we should return an error. 

Example: 

```
lpush l1 e1
hset foo k1 v1
lmpop 2 foo l1 left
>│(error) WRONGTYPE Operation against a key holding the wrong kind of value
```

---

Fixed.